cmd/ignite/run/create.go

```diff
@@ -211,16 +211,22 @@ func applyVMFlagOverrides(baseVM *api.VM, cf *CreateFlags, fs *flag.FlagSet) err
 		baseVM.Spec.Storage = cf.VM.Spec.Storage
 	}
 
-	// Parse the --copy-files flag.
-	baseVM.Spec.CopyFiles, err = parseFileMappings(cf.CopyFiles)
-	if err != nil {
-		return err
+	// TODO: Remove len(cf.CopyFiles) > 0 after fixing TestApplyVMFlagOverrides
+	if fs.Changed("copy-files") || len(cf.CopyFiles) > 0 {
+		// Parse the --copy-files flag.
+		baseVM.Spec.CopyFiles, err = parseFileMappings(cf.CopyFiles)
+		if err != nil {
+			return err
+		}
 	}
 
-	// Parse the given port mappings.
-	baseVM.Spec.Network.Ports, err = meta.ParsePortMappings(cf.PortMappings)
-	if err != nil {
-		return err
+	// TODO: Remove len(cf.PortMappings) > 0 after fixing TestApplyVMFlagOverrides
+	if fs.Changed("ports") || len(cf.PortMappings) > 0 {
```

Comment:
Checking the length for both CopyFiles and PortMappings should be enough to get this right.
"copy-files" and "ports" are not directly taken from the VM spec. They have separate variables for simpler flag friendly types, refer: https://github.com/weaveworks/ignite/blob/3120baf0e4f924a9280d59e7092fda3c57bcccb1/cmd/ignite/cmd/vmcmd/create.go#L71-L72 .

Reply:
I am going to drop the commit https://github.com/weaveworks/ignite/pull/796/commits/c5d81b93f9ba34abc4c6b11afea46ddddbe9573d